### PR TITLE
Basic Web Server Configation

### PR DESCRIPTION
**Pull request by archeious**
Recommitting  j-e-k/poketrainer/217 after rebasing.

### CODE REVIEW

File: web.py [hunk 1]
```diff
@@ -114,5 +115,26 @@ def transfer(username, p_id):
...
+        with open(config_file) as data:
+            load.update(json.load(data))
+    # Read passed in Arguments
+    parser.add_argument("-s", "--hostname", help="Server hostname/IP", default="0.0.0.0")
+    parser.add_argument("-p", "--port", help="Server TCP port number", default=5000)
+    parser.add_argument("-d", "--debug", help="Debug Mode", action='store_true', default=False)
+    config = parser.parse_args()
+    # Passed in arguments shoud trump
+    for key,value in load.iteritems():
+      if key not in config.__dict__ or not config.__dict__[key]:
```
Comment (infinitewarp) [inline web.py]: please use standard 4-space indents

**Comment by infinitewarp** [comment]
I'm glad to see this PR! I was going to do one very much like this, but you beat me to the punch. 👍 

Once you rebase and resolve conflicts, please take a moment to run `tox` (see the readme) against the code and fix any errors it reports. You have a few violations in the current version it should tell you about.

**Comment by archeious** [comment]
tox doesn't work on my machine.  I have fixed the tabbing, rebased again, and created a new PR.  Whats the deal with all the rebasing anyways?

**Comment by Myhorta** [comment]
You don't need to open a new PR each time your code needs to be rebased/merged. You just need to keep the origin(your fork), upstream and your local copy in sync. See [this document](https://help.github.com/articles/syncing-a-fork/), it explains the required steps.

**Pull request closed without merging**